.github/workflows/pr.yaml

```diff
@@ -224,6 +224,11 @@ jobs:
       - name: Run Unit Tests
         run: |
           make test-unit
+      - name: Upload coverage reports to Codecov
+        uses: codecov/codecov-action@e0b68c6749509c5f83f984dd99a76a1c1a231044 #v4.0.1
```

Comment:
Nice!

The latest version is [v4.4.1](https://github.com/codecov/codecov-action/releases/tag/v4.4.1). Is there a reason for not using it? If so, please insert a comment explaining it.